bin/sozo/src/commands/events.rs

```diff
@@ -55,63 +57,78 @@ impl EventsArgs {
     pub fn run(self, config: &Config) -> Result<()> {
         config.tokio_handle().block_on(async {
             let ws = scarb::ops::read_workspace(config.manifest_path(), config)?;
-            let profile_config = ws.load_profile_config()?;
-
             let (world_diff, provider, _) =
                 utils::get_world_diff_and_provider(self.starknet, self.world, &ws).await?;
-
             let provider = Arc::new(provider);
 
-            let from_block = if let Some(world_block) =
-                profile_config.env.as_ref().and_then(|e| e.world_block)
-            {
-                Some(BlockId::Number(world_block))
-            } else {
-                self.from_block.map(BlockId::Number)
-            };
+            // Get latest block number
+            let latest_block = provider.block_number().await?;
 
-            let to_block = self.to_block.map(BlockId::Number);
-            let keys = self
-                .events
-                .map(|e| vec![e.iter().map(|event| starknet_keccak(event.as_bytes())).collect()]);
+            let from_block = match self.from_block {
+                Some(block) => match BlockId::Number(block) {
+                    BlockId::Number(num) => num,
+                    _ => latest_block.saturating_sub(1000),
+                },
+                None => latest_block.saturating_sub(1000),
```

Comment:
Why?

Reply:
The idea was to have a reasonable default window to start fetching event when the event is used. And since you recommend to remove the `world_block` I figured starting from around 24hr from the latest block would be reasonable.

What do you think?

Reply:
Oh I see, so if there's a world block, we take it. If not, you would start 1000 blocks before the latest?

In this case, it may be better starting to `0` actually. Like so, the default behavior is something easy to grasp for user. WDYT?

Reply:
From my understanding the user can specify the `from_block` and `to_block` as a cli argument. If they do not, then we can start from 1000 blocks before the latest.

```rs
#[arg(short, long)]
#[arg(help = "Block number from where to look for events")]
pub from_block: Option<u64>,

#[arg(short, long)]
#[arg(help = "Block number until where to look for events")]
pub to_block: Option<u64>,
```

I don't think it will be better to start from 0 as it will definitely take more time and 100% most of the what ill be fetched is not relevant to us.

If you still want, we can default to `world_block` in a case where a user does not provide a from_block / to_block.

I would also double check later tomorrow, to make sure I am not overlooking something here